packages/astro/src/content/runtime.ts

```diff
@@ -55,9 +55,11 @@ export function createGetCollection({
 		} else if (collection in dataCollectionToEntryMap) {
 			type = 'data';
 		} else {
-			throw new AstroError({
-				...AstroErrorData.CollectionDoesNotExistError,
-				message: AstroErrorData.CollectionDoesNotExistError.message(collection),
+			return zodString().transform((_, ctx) => {
+				ctx.addIssue({
+					code: ZodIssueCode.custom,
+					message: `The collection **${collection}** does not exist. Ensure a collection directory with this name exists.`,
+				});
```

Comment:
I am unfamiliar with zod; what do we do here and why?

Reply:
To be honest, I'm not entirely happy with this. I think this would still lead to an error. 

My initial idea was to use something like `logger.warn` but I couldn't find a logger?

Reply:
You can get a logger instance from each astro hook: https://docs.astro.build/en/reference/integrations-reference/#astrointegrationlogger

Then, you can pass it around and log whatever you want :)

Reply:
Thanks for the hint! I hope this solution is better